src/dodal/devices/oav/pin_tip_detection/oav_with_pin_tip_detection.py

```diff
@@ -0,0 +1,208 @@
+import asyncio
+import time
+from collections import OrderedDict
+from typing import Callable, Optional, Tuple, Type, TypeVar
+
+import numpy as np
+from bluesky.protocols import Descriptor
+from numpy.typing import NDArray
+from ophyd.v2.core import Device, Readable, Reading, SimConverter, SimSignalBackend
+from ophyd.v2.epics import SignalR, SignalRW, epics_signal_r
+
+from dodal.devices.oav.pin_tip_detection.pin_tip_detect_utils import (
+    ArrayProcessingFunctions,
+    MxSampleDetect,
+)
+from dodal.log import LOGGER
+
+T = TypeVar("T")
+
+
+def _create_soft_signal(
+    name: str, datatype: T | Type[T], initial_value: T
+) -> SignalRW[T]:
+    class _Converter(SimConverter):
+        def make_initial_value(self, *args, **kwargs) -> T:
+            return initial_value
+
+    class _SimSignalBackend(SimSignalBackend):
+        async def connect(self) -> None:
+            self.converter = _Converter()
+            self._initial_value = self.converter.make_initial_value()
+            self._severity = 0
+
+            await self.put(None)
+
+    backend = _SimSignalBackend(
+        datatype, f"sim://oav_with_pin_tip_detection_soft_params:{name}"
+    )
+    signal: SignalRW[T] = SignalRW(backend)
+    return signal
+
+
+class PinTipDetection(Readable, Device):
+    def __init__(self, prefix: str, name: str = ""):
+        self._prefix: str = prefix
+        self._name = name
+
+        self.array_data: SignalR[NDArray[np.uint8]] = epics_signal_r(
+            NDArray[np.uint8], f"pva://{prefix}PVA:ARRAY"
+        )
+
+        self.oav_width: SignalR[int] = epics_signal_r(
+            int, f"{prefix}PVA:ArraySize1_RBV"
+        )
+        self.oav_height: SignalR[int] = epics_signal_r(
+            int, f"{prefix}PVA:ArraySize2_RBV"
+        )
+
+        # Soft parameters for pin-tip detection.
+        self.timeout: SignalRW[float] = _create_soft_signal("timeout", float, 10.0)
+        self.preprocess: SignalRW[
+            Callable[[np.ndarray], np.ndarray]
+        ] = _create_soft_signal(
+            "preprocess",
+            Callable[[np.ndarray], np.ndarray],
+            ArrayProcessingFunctions.identity(),
+        )
```

Comment:
Should: This breaks the symmetry with the current way of working in that we send an index to preprocess for which function to use (which I agree is rubbish). We also only ever actually use the Gaussian blur (mostly) or in one occasion on I24 use the dilate. I think maybe better to keep the API here then make a new issue to change it to something better with this in mind